docs/docs/tutorials/jwt_authentication.md

```diff
@@ -0,0 +1,515 @@
+under the hood, so to get a better understanding on how they work, be sure to check their official
+[documentation](https://jwt.io/).
+
+Right, with that brief introduction, lets get that done. Luckily, the Dart ecosystem already have
+a handy package that makes it easy to work with JWT tokens, lets start by adding that dependency to our
+project:
+
+```dart
+dart pub add dart_jsonwebtoken
+```
+
+Next, lets add the following method to our `Authenticator` class:
+
+```dart
+  String generateToken({
+    required String username,
+    required User user,
+  }) {
+    final jwt = JWT(
+      {
+        'id': user.id,
+        'name': user.name,
+        'username': username,
+      },
+    );
+
+    return jwt.sign(SecretKey('123'));
+  }
+```
+
+That will take care of generating a JWT token. Note that we call a method `sign` out of the token,
+passing a secret key, as the name implies, this key is secret and should be kept as such, in this
+tutorial we are keeping it hard coded for the sake of simplicity, but in a real case application,
+be sure to correctly store it and pass it to the code in a way where then will remain secret to outsiders.
+
+The sign method will create a signature out of the data we passed to it, that signature will be part
+of the token and will later on allow us to check if an authentication token that we've received
+is valid and if hasn't been tampered!
+
+Alright, now we need to make a small change in our sign in route, as it should not return the username
+anymore, but rather the token created by this method instead. The route will now look like this:
+
+```dart
+import 'dart:io';
+
+import 'package:authenticated_app/authenticator.dart';
+import 'package:dart_frog/dart_frog.dart';
+
+Future<Response> onRequest(RequestContext context) async {
+  return switch (context.request.method) {
+    HttpMethod.post => _onPost(context),
+    _ => Future.value(
+        Response(statusCode: HttpStatus.methodNotAllowed),
+      ),
+  };
+}
+
+Future<Response> _onPost(RequestContext context) async {
+  final body = await context.request.json() as Map<String, dynamic>;
+  final username = body['username'] as String?;
+  final password = body['password'] as String?;
+
+  if (username == null || password == null) {
+    return Response(statusCode: HttpStatus.badRequest);
+  }
+
+  final authenticator = context.read<Authenticator>();
+
+  final user = authenticator.findByUsernameAndPassword(
+    username: username,
+    password: password,
+  );
+
+  if (user == null) {
+    return Response(statusCode: HttpStatus.unauthorized);
+  } else {
+    return Response.json(
+      body: {
+        'token': authenticator.generateToken(
+          username: username,
+          user: user,
+        ),
+      },
+    );
+  }
+}
+```
+
+Finally, we now need to change the `Authenticator` to verify the signed token instead of just
+checking if there is a user with the username.
```

Comment:
```suggestion
Finally, we now need to change the `Authenticator` to verify the newly signed token, instead of just
checking against a user's username.
```